### PR TITLE
I:R centuries support - Part 1

### PR DESCRIPTION
**Pull request by IohannesIohannium**
(Off-topic, I got myself a working computer.)

- Creates a 150 AD bookmark (currently flagged as unplayable)
- Blocks pilgrimage to pagan faiths that frankly never had a concept of pilgrimage to even begin with
- Adds century scripted triggers
- Adds two new eras (Classical and Late Classical), although images not correctly working and currently having no innovation in them

### CODE REVIEW

**Review by IhateTrains — CHANGES_REQUESTED** [review]

File: ImperatorToCK3/Data_Files/blankMod/output/common/scripted_triggers/00_century_triggers.txt [hunk 1]
```diff
@@ -0,0 +1,58 @@
+﻿1st_century = {
+	NOT = { year = 102 }
+}
+2nd_century = {
+	year = 102
```
[the remark this answers is not in the document]
Comment (IohannesIohannium) [reply ImperatorToCK3/Data_Files/blankMod/output/common/scripted_triggers/00_century_triggers.txt:5]: Centuries start at ....1; for example, the 21st century started at 2001 and will end at 2100.

File: ImperatorToCK3/Data_Files/blankMod/output/common/scripted_triggers/00_century_triggers.txt [hunk 2]
```diff
@@ -0,0 +1,58 @@
+﻿1st_century = {
+	NOT = { year = 102 }
+}
+2nd_century = {
+	year = 102
+	NOT = { year = 202 }
```
[the remark this answers is not in the document]
Comment (IohannesIohannium) [reply ImperatorToCK3/Data_Files/blankMod/output/common/scripted_triggers/00_century_triggers.txt:6]: See above

File: ImperatorToCK3/Data_Files/blankMod/output/localization/english/new_eras_l_english.yml [hunk 3]
```diff
@@ -0,0 +1,7 @@
+﻿l_english:
+
+ culture_era_classical:0 "Classical"
+ culture_era_classical_desc:0 "Classical Era can be best expressed as the glory of the Roman Empire in the centuries before Christendom. Future historians praise this period for its magnificence, forgetting how it is simply the last pang of a dying civilization. Caesarism was simply the last expression of the Greco-Roman culture. Unless..."
+ 
+ culture_era_late_classical:0 "Late Classical"
+ culture_era_late_classical_desc:0 "Lacking a better label, Late Classical refers to the period of troubles that saw the Roman Empire reduced to nothing more than a geographical expression. Devoid of any tradition of culture or faith, it is now a shell and a shadow of its former glory. Meanwhile, the rest of Europe is slowly waking up from a slumber."
```
[the remark this answers is not in the document]
Comment (IohannesIohannium) [reply ImperatorToCK3/Data_Files/blankMod/output/localization/english/new_eras_l_english.yml:7]: My bad, feel free to suggest a better description

File: ImperatorToCK3/Data_Files/blankMod/output/common/scripted_triggers/00_century_triggers.txt (hunk 1 above)
[the remark this answers is not in the document]
Comment (IhateTrains) [reply ImperatorToCK3/Data_Files/blankMod/output/common/scripted_triggers/00_century_triggers.txt:5]: I know, but don't your conditions contradict that?

File: ImperatorToCK3/Data_Files/blankMod/output/common/decisions/10_religious_decisions.txt [hunk 4]
```diff
@@ -0,0 +1,1094 @@
...
+	cooldown = { years = pilgrimage_cooldown_year_amount }
+
+	is_shown = {
+		is_landed = yes
+		NOR = {
+			# These Faiths historically shunned going on pilgrimages, saying they were a waste of time & money.
+			faith = faith:waldensian
+			faith = faith:paulician
+			faith = faith:lollard
+			faith = { is_in_family = rf_pagan } # No Pilgrimage in Paganism
```
[the remark this answers is not in the document]
Comment (IohannesIohannium) [reply ImperatorToCK3/Data_Files/blankMod/output/common/decisions/10_religious_decisions.txt]: Readded pilgrimage for Hellenic and Arab pagans

File: ImperatorToCK3/Data_Files/blankMod/output/localization/ir_welcome_l_english.yml [hunk 5]
```diff
@@ -0,0 +1,5 @@
+﻿l_english:
+ CONVERTERS_TEAM:0 "The Paradox Game Converters Group"
+ welcome_title:0 "Welcome to Crusader Kings III"
+ welcome_desc:0 "You are now playing as... Well, you know the story better than we could. You have ruled for centuries in the Classical Era, dodging the Greeks and the Romans, and now you're here, to witness the rise and fall of a mighty Empire and the civilization that will be born from its ashes. However, let us just add that some bugs may be lingering around and the translations may be wrong in some places. Feel free to report it to us on the Forum. That said, let us go forth!"
+ welcome_ok:0 "Ok"
```
[the remark this answers is not in the document]
Comment (IohannesIohannium) [reply ImperatorToCK3/Data_Files/blankMod/output/localization/ir_welcome_l_english.yml]: Solved

File: ImperatorToCK3/Data_Files/blankMod/output/common/scripted_triggers/00_century_triggers.txt (hunk 1 above)
[the remark this answers is not in the document]
Comment (IohannesIohannium) [reply ImperatorToCK3/Data_Files/blankMod/output/common/scripted_triggers/00_century_triggers.txt]: You're right.

**Comment by IhateTrains** [comment]
Is the bookmark button supposed to look like this?
image

**Comment by IhateTrains** [comment]
Some errors introduced in the PR:
```
[19:24:31][pdx_assert.cpp:568]: Assertion failed: Creation of dynamic token "1st_century" failed, keywords may not start with a digit or '-'.
[19:24:31][lexer.cpp:209]: Creation of dynamic token "1st_century" failed, keywords may not start with a digit or '-'.
[19:24:31][lexer.cpp:209]: Creation of dynamic token "2nd_century" failed, keywords may not start with a digit or '-'.
[19:24:31][lexer.cpp:209]: Creation of dynamic token "3rd_century" failed, keywords may not start with a digit or '-'.
[19:24:31][lexer.cpp:209]: Creation of dynamic token "4th_century" failed, keywords may not start with a digit or '-'.
[19:24:31][lexer.cpp:209]: Creation of dynamic token "5th_century" failed, keywords may not start with a digit or '-'.
[19:24:31][lexer.cpp:209]: Creation of dynamic token "6th_century" failed, keywords may not start with a digit or '-'.
[19:24:31][lexer.cpp:209]: Creation of dynamic token "7th_century" failed, keywords may not start with a digit or '-'.
[19:24:31][lexer.cpp:209]: Creation of dynamic token "8th_century" failed, keywords may not start with a digit or '-'.
[19:24:31][lexer.cpp:209]: Creation of dynamic token "9th_century" failed, keywords may not start with a digit or '-'.
[19:24:31][lexer.cpp:209]: Creation of dynamic token "10th_century" failed, keywords may not start with a digit or '-'.
[19:24:31][lexer.cpp:209]: Creation of dynamic token "11th_century" failed, keywords may not start with a digit or '-'.
[19:24:31][lexer.cpp:209]: Creation of dynamic token "12th_century" failed, keywords may not start with a digit or '-'.
[19:24:31][lexer.cpp:209]: Creation of dynamic token "13th_century" failed, keywords may not start with a digit or '-'.
[19:24:31][lexer.cpp:209]: Creation of dynamic token "14th_century" failed, keywords may not start with a digit or '-'.
[19:24:31][lexer.cpp:209]: Creation of dynamic token "15th_century_or_later" failed, keywords may not start with a digit or '-'.
[19:24:47][jomini_trigger.cpp:871]: Inconsistent trigger scopes (faith vs. religion) in file: common/decisions/10_religious_decisions.txt line: 774
```

**Comment by IohannesIohannium** [comment]
> Is the bookmark button supposed to look like this?
> image

It is how I made it, but anybody who can make it less weird is welcome

**Comment by IohannesIohannium** [comment]
> Some errors introduced in the PR:
> 
> ```
> [19:24:31][pdx_assert.cpp:568]: Assertion failed: Creation of dynamic token "1st_century" failed, keywords may not start with a digit or '-'.
> [19:24:31][lexer.cpp:209]: Creation of dynamic token "1st_century" failed, keywords may not start with a digit or '-'.
> [19:24:31][lexer.cpp:209]: Creation of dynamic token "2nd_century" failed, keywords may not start with a digit or '-'.
> [19:24:31][lexer.cpp:209]: Creation of dynamic token "3rd_century" failed, keywords may not start with a digit or '-'.
> [19:24:31][lexer.cpp:209]: Creation of dynamic token "4th_century" failed, keywords may not start with a digit or '-'.
> [19:24:31][lexer.cpp:209]: Creation of dynamic token "5th_century" failed, keywords may not start with a digit or '-'.
> [19:24:31][lexer.cpp:209]: Creation of dynamic token "6th_century" failed, keywords may not start with a digit or '-'.
> [19:24:31][lexer.cpp:209]: Creation of dynamic token "7th_century" failed, keywords may not start with a digit or '-'.
> [19:24:31][lexer.cpp:209]: Creation of dynamic token "8th_century" failed, keywords may not start with a digit or '-'.
> [19:24:31][lexer.cpp:209]: Creation of dynamic token "9th_century" failed, keywords may not start with a digit or '-'.
> [19:24:31][lexer.cpp:209]: Creation of dynamic token "10th_century" failed, keywords may not start with a digit or '-'.
> [19:24:31][lexer.cpp:209]: Creation of dynamic token "11th_century" failed, keywords may not start with a digit or '-'.
> [19:24:31][lexer.cpp:209]: Creation of dynamic token "12th_century" failed, keywords may not start with a digit or '-'.
> [19:24:31][lexer.cpp:209]: Creation of dynamic token "13th_century" failed, keywords may not start with a digit or '-'.
> [19:24:31][lexer.cpp:209]: Creation of dynamic token "14th_century" failed, keywords may not start with a digit or '-'.
> [19:24:31][lexer.cpp:209]: Creation of dynamic token "15th_century_or_later" failed, keywords may not start with a digit or '-'.
> [19:24:47][jomini_trigger.cpp:871]: Inconsistent trigger scopes (faith vs. religion) in file: common/decisions/10_religious_decisions.txt line: 774
> ```

Solved the dynamic token errors

File: ImperatorToCK3/Data_Files/blankMod/output/common/decisions/10_religious_decisions.txt [hunk 6]
```diff
@@ -0,0 +1,1102 @@
...
+			faith = faith:lollard
+			faith = { # Removing pilgrimage option for pagans who never had such a thing attested
+				is_in_family = rf_pagan
+				NOT = {
+					faith = faith:hellenic_pagan
+					faith = faith:eleusinian_mysteries
+					faith = faith:mithraic_mysteries
+					faith = faith:arabic_pagan
+				}
+			}
```
Comment (IhateTrains) [inline ImperatorToCK3/Data_Files/blankMod/output/common/decisions/10_religious_decisions.txt:781]: I'm still not convinced about the need for the addition, but if it's to be added it needs to be working.
Try
```suggestion
			AND = { # Removing pilgrimage option for pagans who never had such a thing attested
				religion = { is_in_family = rf_pagan }
				NOR = {
					faith = faith:hellenic_pagan
					faith = faith:eleusinian_mysteries
					faith = faith:mithraic_mysteries
					faith = faith:arabic_pagan
				}
			}

```

File: ImperatorToCK3/Data_Files/blankMod/output/localization/english/new_eras_l_english.yml (hunk 3 above)
[the remark this answers is not in the document]
Comment (IhateTrains) [reply ImperatorToCK3/Data_Files/blankMod/output/localization/english/new_eras_l_english.yml:7]: This shouldn't hold the PR back, I will just open an issue to improve the desc.

**Review by IhateTrains — APPROVED** [review]
The script error is fixed.

**Pull request merged**